commit 3cded66330591cfd2554a3fd5edca8859ea365a2
Author: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date:   Tue Nov 24 14:33:20 2020 +0800

    ubifs: Fix error return code in ubifs_init_authentication()
    
    Fix to return PTR_ERR() error code from the error handling case where
    ubifs_hash_get_desc() failed instead of 0 in ubifs_init_authentication(),
    as done elsewhere in this function.
    
    Fixes: 49525e5eecca5 ("ubifs: Add helper functions for authentication support")
    Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
    Reviewed-by: Sascha Hauer <s.hauer@pengutronix.de>
    Signed-off-by: Richard Weinberger <richard@nod.at>

diff --git a/fs/ubifs/auth.c b/fs/ubifs/auth.c
index b93b3cd10bfd..8c50de693e1d 100644
--- a/fs/ubifs/auth.c
+++ b/fs/ubifs/auth.c
@@ -338,8 +338,10 @@ int ubifs_init_authentication(struct ubifs_info *c)
 	c->authenticated = true;
 
 	c->log_hash = ubifs_hash_get_desc(c);
-	if (IS_ERR(c->log_hash))
+	if (IS_ERR(c->log_hash)) {
+		err = PTR_ERR(c->log_hash);
 		goto out_free_hmac;
+	}
 
 	err = 0;